### PR TITLE
corrected Raspian steps

### PR DESCRIPTION
**Pull request by NonCreature0714**
@londoncalling @mstanleyjones

### Proposed changes

Addressing bug #2098, this documentation specifies more clearly what steps are necessary for installing docker-ce on Raspian.

### CODE REVIEW

File: engine/installation/linux/debian.md [hunk 1]
```diff
@@ -146,12 +146,16 @@ from the repository.
...
+    $ echo "deb [arch=armhf] https://apt.dockerproject.org/repo \
+        raspbian-jessie main" | \
+        sudo tee /etc/apt/sources.list.d/docker.list
     ```
 
+    > **Warning**: It is HIGHLY recommended you use
+    > ```bash $ curl -sSL https://get.docker.com | sh ``` in order to install Docker on
+    > Raspberry Pi's!! This will also install bash auto-completion and
+    > appropriately mirror the download depending on your country.
+
```
Comment (londoncalling) [inline engine/installation/linux/debian.md]: @NonCreature0714 just a minor comment to replace the double exclamations with just a period :-) We can leave the all-caps HIGHLY, but I think that and the warning formatting should be enough. (Giving the pros for doing this is great, but if you want to add any particular downside of _not_ using this command, that might be more compelling than the exclamation marks.)

**Comment by londoncalling** [comment]
LGTM - great! I'll merge this and just make a few formatting changes.

 @jcberthon, @sandor-juhasz can you review this for technical content since you all are on the original issue?

**Comment by sandor-juhasz** [comment]
The updated doc looks good, it describes both the convenience-script-based installation and the apt-based manual installation clearly. Thanks for the fix!

**Comment by jcberthon** [comment]
The proposed changes are using the old URL which was used prior to the introduction of Docker CE/EE. It seems that Docker provides a path for those old URL to upgrade to Docker CE, but as far as I can tell this is not the recommended upgrade path on Debian and Ubuntu where the documentation switch from `https://apt.dockerproject.org/repo` to `https://download.docker.com/linux`. This is also hinted by the fact that one would need to install `docker` and not `docker-ce`.

I think the proper installation way on Raspbian is to do as for Debian but not using `add-apt-repository` as I reported:

```
echo "deb [arch=armhf] https://download.docker.com/linux/debian \
   $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list
```

Something was fixed on Docker side, and now I do not see the warning I earlier reported in the issue. Docker run fine with this installation and it even installs the Bash completion files. I will propose a PR for this change. Perhaps someone from Docker could see is instead of using the URL `https://download.docker.com/linux/debian` there is one dedicated to Raspbian or not and correct my PR.

If what I propose above is correct, then mention of the recommended installation via downloading and running a script should be removed, much like it does not exist for Debian nor Ubuntu.

Other than that the PR looks good to me. And my PR will only be a suggestion, I do not mean that this PR was wrong :-)

**Pull request merged**